python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,921 @@
+        if False:
+            # Value to convert peak value to total flux based on flux within psf
+            psfImg = self.diffim.getPsf().computeImage()
+            pkToFlux = np.nansum(psfImg.getArray()) / self.diffim.getPsf().computePeak()
+
+            bg = np.nanmedian(diArr)  # Compute the dipole background (probably very close to zero)
+            startingPk = np.nanmax(diArr) - bg   # use the dipole peak for an estimate.
+            posFlux, negFlux = startingPk * pkToFlux, -startingPk * pkToFlux
+
+            if len(pks) >= 1:
+                posFlux = pks[0].getPeakValue() * pkToFlux
+            if len(pks) >= 2:
+                negFlux = pks[1].getPeakValue() * pkToFlux
+
+        # Use the (flux under the dipole)*5 for an estimate.
+        # Lots of testing showed that having startingFlux be too high was better than too low.
+        startingFlux = np.nansum(np.abs(diArr) - np.nanmedian(np.abs(diArr))) * 5.
+        posFlux = negFlux = startingFlux
+
+        # TBD: set max. flux limit?
+        gmod.set_param_hint('flux', value=posFlux, min=0.1)
+
+        if separateNegParams:
+            # TBD: set max negative lobe flux limit?
+            gmod.set_param_hint('fluxNeg', value=np.abs(negFlux), min=0.1)
+
+        # Fixed parameters (dont fit for them if there are no pre-sub images or no gradient fit requested):
+        # Right now, we use the linear model to fit the background and initialize the parameters.
+        # It might be better to just subtract the linear model from the data and then don't fit the background
+        # again.
+        bgParsPos = bgParsNeg = (0., 0., 0.)
+        if (rel_weight > 0. and fitBgGradient and bgGradientOrder >= 0):
+            pbg = 0.
+            # Fit the gradient to the background (linear model)
+            bgParsPos = bgParsNeg = self.fitBackgroundGradient(source, self.posImage,
+                                                               order=bgGradientOrder)
+            # Generate the gradient and subtract it from the pre-subtraction image data
+            in_x = self._generateXYGrid(bbox)
+            pbg = self.genBgGradientModel(in_x, tuple(bgParsPos))
+            z[1, :] -= pbg
+            z[1, :] -= np.nanmedian(z[1, :])
+            posFlux = np.nansum(z[1, :])
+            gmod.set_param_hint('flux', value=posFlux*1.5, min=0.1)
+
+            if separateNegParams and self.negImage is not None:
+                bgParsNeg = self.fitBackgroundGradient(source, self.negImage, order=bgGradientOrder)
+                pbg = self.genBgGradientModel(in_x, tuple(bgParsNeg))
+            z[2, :] -= pbg
+            z[2, :] -= np.nanmedian(z[2, :])
+            if separateNegParams:
+                negFlux = np.nansum(z[2, :])
+                gmod.set_param_hint('fluxNeg', value=negFlux*1.5, min=0.1)
+
+            if False:  # we have subtracted the background from the images so dont fit anymore (faster!)
```

Comment:
Does this relate to your comment "It might be better to just subtract the linear model from the data and then don't fit the background again"? It sounds as if you are doing exactly that, in which case the first comment is outdated.

If you want to support both techniques then you could provide a config parameter to enable both, and this test could be based on that parameter. If you are sure one is better then please drop the code for the other.